### DISCUSSION TITLE
Public folder and build folder

### DISCUSSION THREAD

**Discussion by RizziSilva** [vercel/next.js#19308] [Help] [answered]
Hi, i'm working on a project that use nextjs for SEO purpose and we have a development environment. Everything is working fine beside the images that we put on our public folder, none of then are working on development environment, but if you run the project locally everything works as expected. One thing that a notice is that none of the files in public folder are being sended to next build folder(.next). Do we have to do something to make static files in public folder avaible on build folder? Current using a custom server approch with express.

**Comment by jamesmosier** [comment, marked as answer]
Hi @RizziSilva. The `public` folder contents do not get copied to the `.next` build folder. They are referenced in-place. Do you have a route setup in your Express server that catches all routes? And are you copying all the contents of the `public` folder to your server? 

```js
server.all('*', (req, res) => {
  return handle(req, res);
});
```

**Reply by RizziSilva** [reply]
@jamesmosier I do have a route like this, but i didn't know that you need to copy the public folder to your server. Going to give a try with this approch, if it works i'll marker your comment as answer.

**Reply by RizziSilva** [reply]
@jamesmosier Hi, i'm current facing another problem with deploying the project to a enverionment, maybe you can help. I was running the application with `next start`, but it ignores our custom server. I have changed to run the application with node `server/app.js`. The problem now is that the release process break with a nextjs error saying that no pages directory was found. We are current copying the .next(created by next build), public(images, etc) and server(contains our custom server file and some others files) folder. Wasn't next suppose to look into `.next` to find the pages directory? Do i have to copy all my project to be able to deploy it?

**Reply by jamesmosier** [reply]
When you build your application the `pages` folder should get built into the `.next` build folder.

If you had a minimal reproduction (repository) of what you're seeing I may be able to assist more.

**Reply by RizziSilva** [reply]
@jamesmosier Hi, i have changed one of my personal next project(used to study) to contain the same structure and scripts, link to it https://github.com/RizziSilva/pokemon-with-next. This project is deployed with vercel here https://pokemon-with-next-la06b8p0e.vercel.app/.
  
The project where we are facing the problem contains more components, pages, etc, but the same structure and the same build files. After running `build-development` we copy `.server`, `.next`, `node_modules` e `public` folders + `package.json` file, and then run `start-development`, the pipeline breaks with a nextjs error saying that no pages directory was found. 
   
This is the folder structure that we have on our server: 
```
opt/
|-    .server/
|     |-  app.js
|-    .next/
|     |- builded-content
|-    public/
|     |-  static-files
|-    node_modules
|     |-  packages
|-    package.json
```

**Reply by RizziSilva** [reply]
I'm current doing same test's and it seens that next is looking for the pages directory in the root of the application, but it has been builded into .next. Do i have to also copy pages folder to my server?

**Reply by RizziSilva** [reply]
@jamesmosier I have figure out the problem, the deploy is now working as expected. Thank you for your time and answers, it was really helpfull. 

The problem was with the `dev` prop you pass too `next({ dev: boolean })` when you are creating a custom server. When you pass `true` to it, next will look to a pages directory in the project root, as it expect you tu be running the application with the source code. Simple passing false when deploying it make everithing works just fine. 

I have misunderstood the use of `dev` in this case. The docs do talk about `dev`, but doesn't said anything about change the way next will look to the folders.
